### PR TITLE
[Merged by Bors] - Interactive account passwords

### PR DESCRIPTION
**Pull request by realbigsean**
## Issue Addressed

#1437

## Proposed Changes

- Make the `--wallet-password` flag optional and creates an interactive prompt if not provided.
- Make the `--wallet-name` flag optional and creates an interactive prompt if not provided.
- Add a minimum password requirement of a 12 character length.
- Update the `--stdin-passwords` flag to `--stdin-inputs` because we have non-password user inputs

### CODE REVIEW

File: account_manager/src/wallet/create.rs [hunk 1]
```diff
@@ -126,30 +139,84 @@ pub fn create_wallet_from_mnemonic(
         .map_err(|e| format!("Unable to open --{}: {:?}", BASE_DIR_FLAG, e))?;
 
     // Create a random password if the file does not exist.
```
Comment (paulhauner) [inline account_manager/src/wallet/create.rs:141]: We might want to move the comment down a few lines, it's not completely accurate here.

File: common/account_utils/src/lib.rs [hunk 2]
```diff
@@ -121,9 +123,26 @@ pub fn read_mnemonic_from_user(use_stdin: bool) -> Result<String, String> {
...
+        if s.ends_with('\r') {
+            s.pop();
+        }
+    }
+}
+
+/// Takes a string password and checks that it meets minimum requirements.
+///
+/// The current minimum password requirement is a 12 character length character length.
+pub fn is_password_sufficiently_complex(password: &[u8]) -> bool {
```
Comment (paulhauner) [inline common/account_utils/src/lib.rs:142]: It's not very important, but if this returned `Result<(), String>` you could avoid repeating `"Please use at least {} characters for your password."` in downstream functions by passing that message via the `Result::Err(String)`.

**Review by paulhauner — CHANGES_REQUESTED** [review]
Looks great! I ran it with and without the interactive prompts without issue. I think it's going to be easier for beginners.

I've just left a few nits, happy to merge after that!

It would be cool to update the docs, since I think this interactive password flow is easier. Perhaps we could just do 4.1 and 4.2  in the [validator guide](https://github.com/sigp/lighthouse/blob/master/book/src/become-a-validator-source.md) here now and do the rest later?

File: common/account_utils/src/lib.rs (hunk 2 above)
Comment (realbigsean) [reply common/account_utils/src/lib.rs:142]: That's smart, easier to update error messages if the password requirements change too.

**Review by paulhauner — APPROVED** [review]
Great!

bors r+

**Pull request closed without merging**